akka-docs/rst/java/code/docs/future/FutureDocTest.java

```diff
@@ -524,6 +531,131 @@ public String call() throws InterruptedException {
     Await.result(result, Duration.create(2, SECONDS));
   }
 
+  @Test
+  public void thenApplyCompletionThread() throws Exception {
+    //#apply-completion-thread
+    final ExecutionContext ec = system.dispatcher();
+    final CountDownLatch countDownLatch = new CountDownLatch(1);
+
+    Future<String> scalaFuture = Futures.future(() -> {
+      assertThat(Thread.currentThread().getName(), containsString("akka.actor.default-dispatcher"));
+      countDownLatch.await(); // do not complete yet
+      return "hello";
+    }, ec);
+
+    CompletionStage<String> fromScalaFuture = FutureConverters.toJava(scalaFuture)
+      .thenApply(s -> { // 1
+        assertThat(Thread.currentThread().getName(), containsString("ForkJoinPool.commonPool"));
```

Comment:
if this would throw it would not fail the test, I guess, since it's not thrown on the test main thread